android/capacitor/src/main/java/com/getcapacitor/BridgeFragment.java

```diff
@@ -0,0 +1,195 @@
+import com.getcapacitor.cordova.MockCordovaInterfaceImpl;
+import com.getcapacitor.cordova.MockCordovaWebViewImpl;
+
+import org.apache.cordova.ConfigXmlParser;
+import org.apache.cordova.CordovaPreferences;
+import org.apache.cordova.PluginEntry;
+import org.apache.cordova.PluginManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Activities that contain this fragment must implement the
+ * {@link BridgeFragment.OnFragmentInteractionListener} interface
+ * to handle interaction events.
+ * Use the {@link BridgeFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class BridgeFragment extends Fragment {
+  private static final String ARG_START_DIR = "startDir";
+
+  private String startDir;
+
+  private OnFragmentInteractionListener mListener;
+
+  private WebView webView;
+  protected Bridge bridge;
+  protected MockCordovaInterfaceImpl cordovaInterface;
+  protected boolean keepRunning = true;
+  private ArrayList<PluginEntry> pluginEntries;
+  private PluginManager pluginManager;
+  private CordovaPreferences preferences;
+  private MockCordovaWebViewImpl mockWebView;
+  private int activityDepth = 0;
+  private String bridgeStartDir;
+
+  private String lastActivityPlugin;
+
+  private List<Class<? extends Plugin>> initialPlugins = new ArrayList<>();
+
+
+  public BridgeFragment() {
+    // Required empty public constructor
+  }
+
+  /**
+   * Use this factory method to create a new instance of
+   * this fragment using the provided parameters.
+   *
+   * @param startDir the directory to serve content from
+   * @return A new instance of fragment BridgeFragment.
+   */
+  public static BridgeFragment newInstance(String startDir) {
+    BridgeFragment fragment = new BridgeFragment();
+    Bundle args = new Bundle();
+    args.putString(ARG_START_DIR, startDir);
+    fragment.setArguments(args);
+    return fragment;
+  }
+
+  protected void init(Bundle savedInstanceState) {
+    loadConfig(this.getActivity().getApplicationContext(), this.getActivity());
+  }
+
+  /**
+   * Load the WebView and create the Bridge
+   */
+  protected void load(Bundle savedInstanceState) {
+    Log.d(LogUtils.getCoreTag(), "Starting BridgeActivity");
+
+    Bundle args = getArguments();
+    String startDir = null;
+
+    if (args != null) {
+      startDir = getArguments().getString(ARG_START_DIR);
+    }
+
+    webView = getView().findViewById(R.id.webview);
+    cordovaInterface = new MockCordovaInterfaceImpl(this.getActivity());
+    if (savedInstanceState != null) {
+      cordovaInterface.restoreInstanceState(savedInstanceState);
+    }
+
+    mockWebView = new MockCordovaWebViewImpl(getActivity().getApplicationContext());
+    mockWebView.init(cordovaInterface, pluginEntries, preferences, webView);
+
+    pluginManager = mockWebView.getPluginManager();
+    cordovaInterface.onCordovaInit(pluginManager);
+
+    preferences = new CordovaPreferences();
```

Comment:
Why do you do this? the preferences are being set on line 125, so this reset them